### ISSUE TITLE
Build failure: python311Packages.pytensor

### ISSUE THREAD

**Issue by charmoniumQ** [NixOS/nixpkgs#274566]
### Steps To Reproduce

1. `nix build -L 'nixpkgs#python311Packages.pytensor'`

### Build log

```
warning: Ignoring setting 'auto-allocate-uids' because experimental feature 'auto-allocate-uids' is not enabled
warning: Ignoring setting 'impure-env' because experimental feature 'configurable-impure-env' is not enabled
bazel-build-jaxlib> Running phase: unpackPhase
bazel-build-jaxlib> unpacking source archive /nix/store/irkiaj3ipwh7k1wgyjnmxpsagnbn4yj1-source
bazel-build-jaxlib> source root is source
bazel-build-jaxlib> Running phase: patchPhase
bazel-build-jaxlib> Running phase: updateAutotoolsGnuConfigScriptsPhase
bazel-build-jaxlib> Running phase: configurePhase
source> trying https://github.com/pymc-devs/pytensor/archive/refs/tags/rel-2.18.2.tar.gz
source>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
source>                                  Dload  Upload   Total   Spent    Left  Speed
source>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
source> 100 3932k  100 3932k    0     0  1384k      0  0:00:02  0:00:02 --:--:-- 1916k
source> unpacking source archive /build/rel-2.18.2.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/16lpxwylijjgg2ry5ql9mpqmggsvxx08-source.drv':
         specified: sha256-uB5VT4wP08pOkHlxdPJTXK4j5ubmf+hk5oHYPM6diHM=
            got:    sha256-tYryKYpu4AZBHExeLQ3TG62fsHogoNz+srFaXYtrDm8=
error (ignored): error: cannot unlink '/tmp/nix-build-bazel-build-jaxlib-0.4.21.drv-2/output/external/llvm-raw': Directory not empty
error: 1 dependencies of derivation '/nix/store/00d2hqghh2hgxznabla9pkkxw18w3z4y-python3.11-pytensor-2.18.2.drv' failed to build
```

### Notify maintainers

@bcdarwin 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.64, NixOS, 24.05 (Uakari), 24.05.20231201.91050ea`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.2`
 - nixpkgs: `/home/sam/.nix-defexpr/channels/nixpkgs`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by bcdarwin** [comment]
I haven't seen this hash mismatch before but it should be fixed in #273654, but the more serious problem is that  `jaxlib` is broken so even after fixing the hash or updating `pytensor` it won't be buildable.

**Comment by kirillrdy** [comment]
on lastest master 3b5083b83662210ef84325b879f5ec9189933822

```
nix-build -A python311Packages.jaxlib
/nix/store/nyrkwc6fvb5k9dx00d78qj0ihp8cc1kf-python3.11-jaxlib-0.4.23

nix-build -A python311Packages.pytensor
/nix/store/3c322953lmldvc7kcdcqi93c6plqw0q2-python3.11-pytensor-2.18.4

```
